src/game/Maps/Map.cpp

```diff
@@ -338,12 +340,10 @@ Map::EnsureGridCreated(GridPair const& p)
 void
 Map::EnsureGridLoadedAtEnter(Cell const& cell, Player* player)
 {
-    NGridType* grid;
+    NGridType* grid = getNGrid(cell.GridX(), cell.GridY());
 
     if (EnsureGridLoaded(cell))
     {
-        grid = getNGrid(cell.GridX(), cell.GridY());
```

Comment:
Won't calling getNGrid before EnsureGridLoaded mean it will always return nullptr if not currently loaded?

Reply:
ye, I can see the problem now. I have reverted that part. Let's never talk about it again :>